packages/core/src/interpreter.ts

```diff
@@ -809,73 +802,68 @@ export class Interpreter<
-        // if (!this.state.activities[activity.type]) {
-        //   break;
-        // }
+        const actorDef = (action as InvokeActionObject<TContext, TEvent>).def;
 
         // Invoked services
-        if (activity.type === ActionTypes.Invoke) {
-          const serviceCreator:
-            | ServiceConfig<TContext, TEvent>
-            | undefined = this.machine.options.services
-            ? this.machine.options.services[activity.src]
-            : undefined;
 
-          const { id, data } = activity;
+        let spawnableCreator = isString(actorDef.src)
+          ? this.machine.options.services
+            ? this.machine.options.services[actorDef.src]
+            : undefined
+          : actorDef.src;
 
-          const autoForward =
-            'autoForward' in activity
-              ? activity.autoForward
-              : !!activity.forward;
+        if (isMachineNode(spawnableCreator)) {
+          spawnableCreator = spawnMachine(spawnableCreator);
+        }
 
-          if (!serviceCreator) {
-            // tslint:disable-next-line:no-console
-            if (!IS_PRODUCTION) {
-              warn(
-                false,
-                `No service found for invocation '${activity.src}' in machine '${this.machine.id}'.`
-              );
-            }
-            return;
+        const { id, data } = actorDef;
+
+        if (!spawnableCreator) {
+          // tslint:disable-next-line:no-console
+          if (!IS_PRODUCTION) {
+            warn(
+              false,
+              `No service found for invocation '${actorDef.src}' in machine '${this.machine.id}'.`
+            );
           }
+          return;
+        }
 
-          const actor = serviceCreator(context, _event.data, {
+        const spawnable = (spawnableCreator as ActorCreator<TContext, TEvent>)(
+          context,
+          _event.data,
+          {
             parent: this as any,
             id,
             data,
             _event
-          });
-
-          if (autoForward) {
-            this.forwardTo.add(id);
           }
+        );
 
-          this.children.set(id, actor);
-
-          const childIndex = this.state.children.findIndex(
-            child => child.id === id
-          );
+        const actor = actorFrom(spawnable, id, this as Actor);
 
-          this.state.children[childIndex] = actor;
+        // Mutate the deferred actor reference
+        if (action.actor) {
+          Object.assign(action.actor, actor);
+        }
 
-          this.state.children[childIndex].meta = {
-            ...this.state.children[childIndex].meta,
-            ...activity
-          };
-        } else {
-          this.spawnActivity(activity);
+        if (actorDef.autoForward) {
+          this.forwardTo.add(id);
         }
 
+        this.children.set(id, actor);
+
+        this.state.children[id] = actor;
+
+        this.state.children[id].meta = {
+          ...this.state.children[id].meta,
+          ...actorDef
```

Comment:
is this definition useful anyhow for consumers?  the definition consists solely of `{id,src}`, consumers already have to know the ID to even access this object - so the question is, how `src` can be utilized?

Reply:
Rehydration

Reply:
Could we add a rehydration test that would use it? Would probably be good to include something like this also in the docs.